rafx-renderer/src/swapchain_handling.rs

```diff
@@ -127,20 +125,20 @@ impl<'a> RafxSwapchainEventListener for SwapchainHandler<'a> {
         _device_context: &RafxDeviceContext,
         _swapchain: &RafxSwapchain,
     ) -> RafxResult<()> {
-        let mut guard = self.renderer.inner.lock().unwrap();
-        let renderer = &mut *guard;
-
         log::debug!("renderer swapchain destroyed");
 
-        let mut render_resources = renderer.render_thread.render_resources().lock().unwrap();
-        render_resources.remove::<SwapchainResources>();
+        let mut swapchain_render_resource = self
+            .renderer
+            .render_resources
+            .fetch_mut::<SwapchainRenderResource>();
+        swapchain_render_resource.clear_swapchain();
 
         //TODO: Explicitly remove the images instead of just dropping them. This prevents anything
         // from accidentally using them after they've been freed
-        //swapchain_resources.swapchain_images.clear();
+        //swapchain_render_resource.swapchain_images.clear();
```

Comment:
Can this commented code be removed or is the `TODO` still relevant?

Reply:
I think this is still worth doing at some point